### PR TITLE
Fix spec for `out` contracts without identifier

### PR DESCRIPTION
**Pull request by ntrel**
* Non-void functions can use `out` without result variable too
* Replace HTML tags.
* Minor tweaks.

### CODE REVIEW

File: spec/contracts.dd [hunk 1]
```diff
@@ -99,18 +97,22 @@ body
     return cast(long)std.math.sqrt(cast(real)x);
```
Comment (wilzbach) [inline spec/contracts.dd]: Btw shouldn't this page not also be updated in regards of [DIP1003](http://forum.dlang.org/post/esjclvbebspqkekjmwki@forum.dlang.org)?

File: spec/contracts.dd [hunk 2]
```diff
@@ -122,8 +124,18 @@ body
     ...
 }
 ------
-	In an out statement, $(I result) is initialized and set to the
-	return value of the function.
+	$(P When the return type is not `void`, and no identifier is
+	declared, a variable named `__result` is defined by the compiler
+	for the `out` contract and initialized to the return value of the
+	function:)
```
Comment (wilzbach) [inline spec/contracts.dd]: Isn't this just the current behavior of DMD?

Comment (ntrel) [inline spec/contracts.dd]: OK, maybe `__result` shouldn't be documented then. @zachthemystic quoted use of it to me [here](https://github.com/dlang/DIPs/pull/79#discussion_r126827866), I assumed it was a language feature.

Comment (ntrel) [inline spec/contracts.dd]: Removed `__result` docs.

File: spec/contracts.dd (hunk 1 above)
Comment (ntrel) [inline spec/contracts.dd]: Maybe @MetaLang wants to update contracts.html and grammar.html? Anyway, AIUI the current stable compiler doesn't support it yet.

Comment (MetaLang) [inline spec/contracts.dd]: I'll put it on my todo list.

File: spec/contracts.dd [hunk 3]
```diff
@@ -122,8 +127,6 @@ body
     ...
 }
 ------
-	In an out statement, $(I result) is initialized and set to the
-	return value of the function.
```
Comment (ntrel) [inline spec/contracts.dd]: Removed as it repeats what's on line 84.

File: spec/contracts.dd (hunk 1 above)
Comment (MetaLang) [inline spec/contracts.dd]: And here they are:
https://github.com/dlang/dlang.org/pull/1828
https://github.com/dlang/dlang.org/pull/1829

**Comment by ntrel** [comment]
@CyberShadow Any idea why DAutoTest failed on the last commit?

**Comment by CyberShadow** [comment]
> Any idea why DAutoTest failed on the last commit?

code.dlang.org was down.

**Comment by ntrel** [comment]
@wilzbach thanks!

**Pull request merged**